org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WStringEditor.java

```diff
@@ -312,6 +316,26 @@ else if (WEditorPopupMenu.REQUERY_EVENT.equals(evt.getContextEvent()))
 		{
 			actionRefresh();
 		}
+		else if (WEditorPopupMenu.RESET_EVENT.equals(evt.getContextEvent()))
+		{
+			adwindowContent = findADWindowContent();
+
+			if (adwindowContent.isPendingChanges())
+				adwindowContent.getActiveGridTab().dataSave(true);
+
+			boolean needSwitch = !adwindowContent.getADTab().getSelectedTabpanel().isGridView();
+			if (needSwitch)
+				adwindowContent.getADTab().getSelectedTabpanel().switchRowPresentation();
+
+			GenericPO gpo = new GenericPO(gridTab.getTableName(), Env.getCtx(), gridTab.getRecord_ID());
+			gpo.set_ValueNoCheck(gridField.getColumnName(), null);
+			gpo.saveEx();
+
+			gridTab.dataRefresh();
+
+			if (needSwitch)
+				adwindowContent.getADTab().getSelectedTabpanel().switchRowPresentation();
```

Comment:
this is not right, should use the editor to reset value to null or empty.

Reply:
I agree, but using setValue (null) and saving record doesn't empty the field, but set it to the "default" value ("#000000").
And I can't find where this is set.
The field is "created" on DefaultEditorFactory.getEditor at line 100 ; but I can't find any clue there.
Any idea ?

Reply:
done